druntime/src/core/internal/backtrace/dwarf.d

```diff
@@ -263,6 +268,141 @@ int processCallstack(Location[] locations, const(ubyte)[] debugLineSectionData,
     return 0;
 }
 
+version (Darwin) {
+    /**
+     * Resolve the addresses of `locations` using `atos` (executable that ships with XCode)
+     *
+     * Spawns a child process that calls `atos`. Communication is through stdin/stdout pipes.
+     *
+     * After this function successfully completes, `locations` will contain
+     * file / lines informations.
+     *
+     * The lifetime of the `Location` data surpases function return (strndup is used).
+     *
+     * Params:
+     * locations = The locations to resolve
+     */
+    private void resolveAddressesWithAtos(Location[] locations) @nogc nothrow
+    {
+        import core.stdc.stdio : fclose, fflush, fgets, fprintf, printf, snprintf;
+        import core.stdc.stdlib : exit;
+        import core.sys.posix.stdio : fdopen;
+        import core.sys.posix.unistd : close, dup2, execlp, fork, getpid, pipe;
+        // Create in/out pipes to communicate with the forked exec
+        int[2] dummy_pipes; // these dummy pipes are there to prevent funny issues when stdin/stdout is closed and pipe returns id 0 or 1
+        int[2] pipes_to_atos;
+        int[2] pipes_from_atos;
+        if ( pipe(dummy_pipes) < 0 || pipe(pipes_to_atos) < 0 || pipe(pipes_from_atos) < 0 ) {
```

Comment:
```suggestion
        if ( pipe(dummy_pipes) < 0 || pipe(pipes_to_atos) < 0 || pipe(pipes_from_atos) < 0 ) 
        {
```

Reply:
normally I would but I want this to be identical with the LDC version